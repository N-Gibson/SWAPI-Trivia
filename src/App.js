import React, { Component } from 'react';
import './App.css';
import { apiCalls } from './apiCalls'

class App extends Component {
  constructor() {
    super();
    this.state = {
      movies: [],
      loading: true
    }
  }

  componentDidMount() {
    const swapiFilmsUrl = 'https://swapi.co/api/films';
    apiCalls(swapiFilmsUrl)
      .then(data => this.setState({ movies: data, loading: false }))
  }

  render() {
    console.log(this.state)
    return (
      <main>
        hello
      </main>
    )
  }
}

export default App;