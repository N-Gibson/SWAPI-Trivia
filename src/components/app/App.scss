@font-face {
  font-family: "Star-Jedi";
  src: url('../../fonts/starjedi-webfont.woff') format('woff')
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

main {
  background-image: url('https://media.giphy.com/media/hsITL773u90X6KprRR/giphy.gif');
  background-size: auto;
}