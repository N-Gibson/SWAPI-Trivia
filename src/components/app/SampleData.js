const SampleData = [{
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }]

  export default SampleData