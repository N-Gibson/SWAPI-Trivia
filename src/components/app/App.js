import React, { Component } from 'react';
import './App.css';
import { apiCalls } from './apiCalls'

class App extends Component {
  constructor() {
    super();
    this.state = {
      movies: [],
      loading: true
    }
  }

  componentDidMount() {
    const swapiFilmsUrl = 'https://swapi.co/api/films';
    apiCalls(swapiFilmsUrl)
      .then(res => console.log("DATA", res))
    // console.log("Head", apiCalls(swapiFilmsUrl))
  }

  render() {
    return (
      <main>
        hello
      </main>
    )
  }
}

export default App;