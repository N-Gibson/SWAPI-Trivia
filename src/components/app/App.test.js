import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import SampleData from './SampleData';
// jest.mock('./App');

describe('App', () => {
  let wrapper, movies;

  beforeEach(() => {
    wrapper = shallow(<App />)
      movies = [{
        episode_id: 1,
        image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
        release_date: "1999-05-19",
        title: "The Phantom Menace",
        trailer: 'https://www.youtube.com/watch?v=bD7bpG-zDJQ',
        opening_crawl: "Turmoil has engulfed the Galactic Republic. The taxation of trade routes to outlying star  systems is in dispute. Hoping to resolve the matter with a blockade of deadly battleships, the greedy Trade Federation has stopped all shipping to the small planet of Naboo. While the Congress of the Republic endlessly debates this alarming chain of events, the Supreme Chancellor has secretly dispatched two Jedi Knights, the guardians of peace and justice in the galaxy, to settle the conflict....",
        characters: [{
          films: ["The Phantom Menace"],
          homeworld: ["Mars", "234567890"],
          id: Date.now(),
          isFavorite: false,
          name: "Qui-Gon Jinn",
          species: "Human"
        }]
      }]
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to change order color', () => {
    //setup
    const mockEvent = {
      target: {
        parentNode: {
          className: 'jedi-btn'
        }
      }
    }
    const expectedOrderColor = 'jediColor';
    const expectedOrderRank =  ['Padawan', 'Jedi Master', 'Grand Master of the Jedi'];
    const expectedRank = 'Padawan';
    //execution
    wrapper.instance().handleOrderColor(mockEvent);
    //expectation
    expect(wrapper.state('orderColor')).toEqual(expectedOrderColor);
    expect(wrapper.state('orderRank')).toEqual(expectedOrderRank);
    expect(wrapper.state('rank')).toEqual(expectedRank);
  });

  it('should handle form change', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'value'
      }
    }
    wrapper.instance().handleFormChange(mockEvent)
    expect(wrapper.state('name')).toEqual('value')
  });

  it('should set rank', () => {
    const mockEvent = {
      target: {
        value: 'value'
      }
    }
    wrapper.instance().setRank(mockEvent)
    expect(wrapper.state('rank')).toEqual('value')
  });

  it.skip('should be able to favorite a new character', () => {
    //setup
    const mockEvent = {
      target: {
        parentNode: {
          children: [{
            innerText: 'Qui-Gon Jinn'
          }]
        }
      }
    }
    wrapper.instance().favoriteNewCharacter(mockEvent)
    //expectation
    expect(wrapper.instance().favoriteNewCharacter).toHaveBeenCalled()
  })

  it.skip('should check that apiCalls is fired', () => {
    //setup
    // const mockUrl = 'url'
    // const mockApiCalls = jest.fn(mockUrl)
    //execution
    wrapper.instance().componentDidMount()
    //expectation
    expect(wrapper.instance().componentDidMount).toHaveBeenCalled()
  });
})