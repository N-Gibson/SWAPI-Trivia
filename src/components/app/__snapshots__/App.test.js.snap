// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id/characters"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/characters/favorites"
    render={[Function]}
  />
</main>
`;
